### PR TITLE
Update kustomize and controller-gen, fix CRD_OPTIONS, fix kuttl tests

### PR DESCRIPTION
**Pull request by burmanm**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by burmanm** [comment]
@Miles-Garnsey

File: Makefile [hunk 1]
```diff
@@ -41,7 +41,7 @@ BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
 # Image URL to use all building/pushing image targets
 IMG ?= $(IMAGE_TAG_BASE):latest
 # Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
-CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
+CRD_OPTIONS ?= "crd"
```
Comment (jsanda) [inline Makefile:44]: Does `preserveUnknownFields` default to `false`?

**Review by jsanda — COMMENTED** [review]
Shouldn't this PR include the update CRD manifests?

Can you also create an issue to reference?

File: Makefile (hunk 1 above)
Comment (burmanm) [reply Makefile:44]: Yes and those options are no longer present in newer controller-gen versions. Leaving those options makes the build fail.

**Comment by burmanm** [comment]
@Miles-Garnsey I'll move this discussion here for just the kustomize version, but this applies to PR #286 requirements. Requiring Kustomize 4.4.1 will break usage of `kubectl apply -k` for most users. The default version for some cloud providers is 1.21, but people might use even older ones.

1.21 kubectl ships with Kustomize 4.0.5 integrated. https://github.com/kubernetes-sigs/kustomize#kubectl-integration

I think we should keep the ability to install k8ssandra-operator with just `kubectl apply -k` instead of requiring user to download Kustomize separately before installation. And we can't expect every user to be running the newest kubectl client especially if they're using a server based of 1.21. kubectl will actually even complain about the version skew. 

So if those replacement changes require newer version of Kustomize, we might need to think about doing them in post-1.0 and instead now concentrate on some approach (ugly or not) that would work with older kustomize version also.

File: .github/workflows/kuttl_tests.yaml [hunk 2]
```diff
@@ -75,6 +75,16 @@ jobs:
     steps:
     - name: Check out code into the Go module directory
       uses: actions/checkout@v2
+    - name: Set up Go 1.16
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.16
+    - uses: actions/cache@v2
```
Comment (adutra) [inline .github/workflows/kuttl_tests.yaml:82]: Could you give a name to this step? Otherwise it shows up as "Run actions/cache@v2".

**Review by adutra — APPROVED** [review]

File: .github/workflows/kuttl_tests.yaml (hunk 2 above)
Comment (burmanm) [reply .github/workflows/kuttl_tests.yaml:82]: Done. Although I'm not sure what value the name has in this case ;)

File: .github/workflows/kuttl_tests.yaml [hunk 3]
```diff
@@ -75,6 +75,17 @@ jobs:
     steps:
     - name: Check out code into the Go module directory
       uses: actions/checkout@v2
+    - name: Set up Go 1.16
```
Comment (Miles-Garnsey) [inline .github/workflows/kuttl_tests.yaml:78]: Thanks for identifying and fixing this @burmanm, I can see what you meant about these tests using the wrong Go version now.

**Comment by jsanda** [comment]
@burmanm I think we can ignore the test failures if that is holding up merging.

**Pull request merged**